### PR TITLE
Update for SDK 0.2.X and fixup meta files

### PR DESCRIPTION
**Pull request by jmrossy**
- Update code to work with SDK 0.2.X
- Fixup package.json
- Set up linting
- Add License

TODO:
- [x] Fix tests
- [x] Update lib with type and export fixes

### CODE REVIEW

File: src/deploy/scripts/check.ts [hunk 1]
```diff
@@ -11,14 +20,32 @@ async function check() {
...
-  await yoChecker.check();
-  yoChecker.expectEmpty();
+  const contractsMap = buildContracts(
+    addresses,
+    helloWorldFactories,
+  ) as ChainMap<ChainName, HelloWorldContracts>;
+  // @ts-ignore TODO fix fromEnvironment param type
+  const core = AbacusCore.fromEnvironment('test', multiProvider);
+  const interchainGasCalculator = new InterchainGasCalculator(
+    // @ts-ignore TODO fix multiProvider type issues
```
Comment (jmrossy) [inline src/deploy/scripts/check.ts:30]: @yorhodes Any idea why the type of the multiprovider created above doesn't work here?

Comment (yorhodes) [reply src/deploy/scripts/check.ts:30]: not really sure
if `core` is `AbacusCore<'test1' | 'test2' | 'test3'>` and `multiProvider` is `MultiProvider<'test1' | 'test2' | 'test3'>` it should work

File: src/deploy/scripts/utils.ts [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+import '@nomiclabs/hardhat-ethers';
+import * as fs from 'fs';
+import * as path from 'path';
 
-export async function getEnvironmentConfig(
-  environment: keyof typeof environments,
-): Promise<HelloWorldEnvironmentConfig> {
-  return environments[environment];
+// TODO move to deployer utils
+export function writeContracts(
```
[the remark this answers is not in the document]
Comment (yorhodes) [reply src/deploy/scripts/utils.ts:12]: honestly I don't think we need this file in the template

File: src/deploy/config.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export const testNetworks = [Chains.test1, Chains.test2, Chains.test3] as const;
+export type TestNetworks = typeof testNetworks[number];
+export const testConfigs = {
+  test1: chainConnectionConfigs.test1,
+  test2: chainConnectionConfigs.test2,
+  test3: chainConnectionConfigs.test3,
+};
+
+export function getConfigMap(signerAddress: string) {
```
Comment (yorhodes) [inline src/deploy/config.ts:18]:
```suggestion
export function getConfigMap(signerAddress: string): ChainMap<TestNetworks, RouterConfig> {
```

File: src/deploy/config.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { EnvironmentConfig, RouterConfig } from '@abacus-network/deploy';
+import { chainConnectionConfigs, ChainName, Chains } from '@abacus-network/sdk';
+import { addresses } from '../sdk/environments/test';
+
+// TODO remove all this boilerplate
+export type HelloWorldEnvironmentConfig = EnvironmentConfig<ChainName> & {
+  config: RouterConfig;
+};
```
Comment (yorhodes) [inline src/deploy/config.ts:8]: I think we can remove this?

File: src/deploy/deploy.ts [hunk 5]
```diff
@@ -1,62 +1,28 @@
...
-    super(multiProvider, crossConfigMap, core);
-  }
-
-  async deployContracts(
-    network: Networks,
-    config: HelloWorldConfig,
-  ): Promise<HelloWorldAddresses> {
-    const dc = this.multiProvider.getDomainConnection(network);
-    const signer = dc.signer!;
+  // TODO remove need for this boilerplate
```
Comment (yorhodes) [inline src/deploy/deploy.ts:15]: there's a couple in-flight changes upstream that maybe we should land today before publishing

File: src/deploy/scripts/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmrossy) [reply src/deploy/scripts/utils.ts:12]: I agree, it should be in utils somewhere. I wasn't sure how much we wanted to change the current tests. Should we remove the file writing entirely?

**Review by yorhodes — APPROVED** [review]
looks great

**Pull request merged**